api/payments.js

```diff
@@ -448,12 +442,15 @@ function getAccountPayments(request, response, next) {
   };
 
   function attachResourceId(transactions, callback) {
-    async.map(transactions, function(payment, async_map_callback) {
+    async.map(transactions, function(paymentResult, async_map_callback) {
+      var hash = paymentResult.hash;
+      var payment = paymentResult.payment;
+
       if (!payment) {
         async_map_callback(new Error('No payment found'));
```

Comment:
This isn't client input, so it would only catch that `formatTransactions` is doing the wrong thing, which would be caught in the test. I'd rather remove the check for `payment`